src/libcrun/linux.c

```diff
@@ -1611,7 +1611,16 @@ libcrun_create_dev (libcrun_container_t *container, int devfd, int srcfd,
   int rootfsfd = get_private_data (container)->rootfsfd;
   const char *rootfs = get_private_data (container)->rootfs;
   size_t rootfs_len = get_private_data (container)->rootfs_len;
-  const char *rel_dev = relative_path_under_dev (device->path);
+  // Normalize path by removing trailing slash if it exists.
+  size_t path_len = strlen (device->path);
+  char normalized_path[path_len + 1];
+  memcpy (normalized_path, device->path, path_len);
+
+  if (normalized_path[path_len - 1] == '/')
+    {
+      normalized_path[path_len - 1] = '\0';
+    }
```

Comment:
This won't work if there are two slashes at the end (which is otherwise OK).

Reply:
Basically need to replace `if` with `while`.

Reply:
Having said that, I'd rather fix it in e.g. cri-o. Asking to create a block device with a slash at the end is somewhat weird.

OTOH runc for some reason works.

Reply:
I agree CRI-O is handing crun garbage. I am fine with fixing this, but the real source of the problem is a device with a trailing "/" is and invalid device.

Reply:
>OTOH runc for some reason works.

https://github.com/containers/crun/pull/1517#issuecomment-2284500856

Reply:
@haircommander, should we try to handle this in CRI-O before it even gets to a runtime?

Reply:
we could for sure, the code would be easier to write in go

Reply:
I wouldn't mind handling this in CRI-O. The only caveat is that if some other CRI implementation hits the same issue, then they might question the crun code because this case is already handled in runc

Reply:
I think that's a "cross the bridge when we get there" scenario. This was opened up because it's a bug in the test and it's arguable whether runc should even silently allow it (it's undefined in the runtime spec).

Reply:
Yeah, I would prefer, personally, that the user receive feedback earlier that their path is not conformant (per what @rhatdan said above) rather than us correcting it behind the scenes, so to speak.

If not, then we need to clean the paths up in runc and crun, too. Perhaps even in CRI-O as the first line of defence against malformed values.